src/cockpit/channels/filesystem.py

```diff
@@ -214,10 +217,16 @@ def do_done(self):
 
             try:
                 os.rename(self._temppath, self._path)
-            except OSError:
+            except OSError as exc:
                 # ensure to not leave the temp file behind
                 self.unlink_temppath()
-                raise
+                if isinstance(exc, FileNotFoundError):
```

Comment:
I feel like this is worse than duplicating the `unlink_temppath()`.  How about a `finally:` block or something else?

Reply:
ok! finally: doesn't work, but duplicating the unlink_temppath() doesn't hurt much.